src/ray/stats/metric.cc

```diff
@@ -0,0 +1,100 @@
+#include "ray/stats/metric.h"
+
+namespace ray {
+
+namespace stats {
+
+std::unique_ptr<TagsType> &GetGlobalTagsPtr() {
+  static std::unique_ptr<TagsType> global_tags_ptr(nullptr);
+  return global_tags_ptr;
```

Comment:
Can we use a singleton class to store `global_tags_ptr`, `is_stats_disabled`, and other static fields?

Also, can we just store a `global_tags` object, instead of a pointer?